src/kernel/src/idt/mod.rs

```diff
@@ -68,6 +68,23 @@ impl<T> Idt<T> {
         Ok((value, id))
     }
 
+    /// See `id_rlock` on the PS4 for a reference.
+    pub fn get(&self, id: usize, ty: Option<u16>) -> Option<&Entry<T>> {
```

Comment:
Taking an Option\<u16\> Seems weird to me. I think we should take a u16 and then below this, do 
let set  = self.sets.get(i / 0x80)?;